<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
 <head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <title>Retorna Exception anterior</title>
<link media="all" rel="stylesheet" type="text/css" href="styles/03e73060321a0a848018724a6c83de7f-theme-base.css" />
<link media="all" rel="stylesheet" type="text/css" href="styles/03e73060321a0a848018724a6c83de7f-theme-medium.css" />

 </head>
 <body class="docs"><div class="navbar navbar-fixed-top">
  <div class="navbar-inner clearfix">
    <ul class="nav" style="width: 100%">
      <li style="float: left;"><a href="exception.getmessage.html">« Exception::getMessage</a></li>
      <li style="float: right;"><a href="exception.getcode.html">Exception::getCode »</a></li>
    </ul>
  </div>
</div>
<div id="breadcrumbs" class="clearfix">
  <ul class="breadcrumbs-container">
    <li><a href="/">PHP-EDUC</a></li><li><a href="/php-doc/">Manual</a></li>
    <li><a href="class.exception.html">Exception</a></li>
    <li>Retorna Exception anterior</li>
  </ul>
</div>
<div id="layout">
  <div id="layout-content"><div id="exception.getprevious" class="refentry">
 <div class="refnamediv">
  <h1 class="refname">Exception::getPrevious</h1>
  <p class="verinfo">(PHP 5 &gt;= 5.3.0, PHP 7, PHP 8)</p><p class="refpurpose"><span class="refname">Exception::getPrevious</span> &mdash; <span class="dc-title">Retorna Exception anterior</span></p>

 </div>

 <div class="refsect1 description" id="refsect1-exception.getprevious-description">
  <h3 class="title">Descrição</h3>
  <div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="methodname"><strong>Exception::getPrevious</strong></span>(): <span class="type"><span class="type">?</span><span class="type"><a href="class.throwable.html" class="type Throwable">Throwable</a></span></span></div>

  <p class="para rdfs-comment">
   Retorna a Exception anterior (o terceiro valor passado
   no construtor <span class="methodname"><a href="exception.construct.html" class="methodname">Exception::__construct()</a></span>).
  </p>
 </div>


 <div class="refsect1 parameters" id="refsect1-exception.getprevious-parameters">
  <h3 class="title">Parâmetros</h3>
  <p class="para">Esta função não possui parâmetros.</p>
 </div>


 <div class="refsect1 returnvalues" id="refsect1-exception.getprevious-returnvalues">
  <h3 class="title">Valor Retornado</h3>
  <p class="para">
   Retorna a <span class="classname"><a href="class.exception.html" class="classname">Exception</a></span> anterior se disponível,
   ou senão <strong><code>null</code></strong>.
  </p>
 </div>


 <div class="refsect1 examples" id="refsect1-exception.getprevious-examples">
  <h3 class="title">Exemplos</h3>
  <p class="para">
   <div class="example" id="example-365">
    <p><strong>Exemplo #1 <span class="methodname"><strong>Exception::getPrevious()</strong></span> example</strong></p>
    <div class="example-contents"><p>
     Iterando e imprimindo a pilha de erro.
    </p></div>
    <div class="example-contents">
<div class="phpcode"><code><span style="color: #000000">
<span style="color: #0000BB">&lt;?php<br /></span><span style="color: #007700">class </span><span style="color: #0000BB">MyCustomException </span><span style="color: #007700">extends </span><span style="color: #0000BB">Exception </span><span style="color: #007700">{}<br /><br />function </span><span style="color: #0000BB">doStuff</span><span style="color: #007700">() {<br />    try {<br />        throw new </span><span style="color: #0000BB">InvalidArgumentException</span><span style="color: #007700">(</span><span style="color: #DD0000">"Primeiro erro lançado."</span><span style="color: #007700">, </span><span style="color: #0000BB">112</span><span style="color: #007700">);<br />    } catch(</span><span style="color: #0000BB">Exception $e</span><span style="color: #007700">) {<br />        throw new </span><span style="color: #0000BB">MyCustomException</span><span style="color: #007700">(</span><span style="color: #DD0000">"Segundo erro lançado."</span><span style="color: #007700">, </span><span style="color: #0000BB">911</span><span style="color: #007700">, </span><span style="color: #0000BB">$e</span><span style="color: #007700">);<br />    }<br />}<br /><br /><br />try {<br />    </span><span style="color: #0000BB">doStuff</span><span style="color: #007700">();<br />} catch(</span><span style="color: #0000BB">Exception $e</span><span style="color: #007700">) {<br />    do {<br />        </span><span style="color: #0000BB">printf</span><span style="color: #007700">(</span><span style="color: #DD0000">"%s:%d %s (%d) [%s]\n"</span><span style="color: #007700">, </span><span style="color: #0000BB">$e</span><span style="color: #007700">-&gt;</span><span style="color: #0000BB">getFile</span><span style="color: #007700">(), </span><span style="color: #0000BB">$e</span><span style="color: #007700">-&gt;</span><span style="color: #0000BB">getLine</span><span style="color: #007700">(), </span><span style="color: #0000BB">$e</span><span style="color: #007700">-&gt;</span><span style="color: #0000BB">getMessage</span><span style="color: #007700">(), </span><span style="color: #0000BB">$e</span><span style="color: #007700">-&gt;</span><span style="color: #0000BB">getCode</span><span style="color: #007700">(), </span><span style="color: #0000BB">get_class</span><span style="color: #007700">(</span><span style="color: #0000BB">$e</span><span style="color: #007700">));<br />    } while(</span><span style="color: #0000BB">$e </span><span style="color: #007700">= </span><span style="color: #0000BB">$e</span><span style="color: #007700">-&gt;</span><span style="color: #0000BB">getPrevious</span><span style="color: #007700">());<br />}<br /></span><span style="color: #0000BB">?&gt;</span>
</span>
</code></div>
    </div>

    <div class="example-contents"><p>O exemplo acima produzirá
algo semelhante a:</p></div>
    <div class="example-contents screen">
<div class="cdata"><pre>
/home/bjori/ex.php:8 Segundo erro lançado. (911) [MyCustomException]
/home/bjori/ex.php:6 Primeiro erro lançado. (112) [InvalidArgumentException]
</pre></div>
    </div>
   </div>
  </p>
 </div>


 <div class="refsect1 seealso" id="refsect1-exception.getprevious-seealso">
  <h3 class="title">Veja Também</h3>
  <p class="para">
   <ul class="simplelist">
    <li class="member"><span class="methodname"><a href="throwable.getprevious.html" class="methodname" rel="rdfs-seeAlso">Throwable::getPrevious()</a> - Returns the previous Throwable</span></li>
   </ul>
  </p>
 </div>

</div></div></div></body></html>